archiver.go

```diff
@@ -497,6 +529,84 @@ SELECT rec.visibility, row_to_json(rec) FROM (
 	ORDER BY created_on ASC, id ASC) rec; 
 `
 
+const lookupPurgedBroadcasts = `
+SELECT row_to_json(rec) FROM (
+	SELECT
+	NULL AS id,
+	mb.id AS broadcast,
+	jsonb_build_object('uuid', c.uuid, 'name', c.name) as contact,
+	COALESCE(mb.text->c.language, mb.text->$2, mb.text->'base') AS text,
+	NULL AS urn,
+	NULL AS channel,
+	'out' AS direction,
+	'flow' AS type,
+	'wired' AS status,
+	'visible' AS visibility,
+	'[]'::jsonb AS attachments,
+	'[]'::jsonb AS labels,
+	mb.created_on AS created_on,
+	mb.created_on AS sent_on
+	FROM msgs_broadcast_recipients br
+	JOIN LATERAL (select uuid, name, language FROM contacts_contact cc WHERE cc.id = br.contact_id AND cc.is_test = FALSE) AS c ON TRUE
+	LEFT JOIN msgs_broadcast mb ON br.broadcast_id = mb.id
+	WHERE br.broadcast_id = ANY (
+	   ARRAY(
+		SELECT id FROM msgs_broadcast 
+		WHERE org_id = $1 AND created_on > $3 AND created_on < $4 AND purged = TRUE
+		ORDER by created_on, id
```

Comment:
@dodobas any ideas on making this part faster? I don't think INs are recommended but I couldn't figure out a better way of doing this that was performant. (look at jan/feb on rapid for org 2 to test)